web/src/containers/Expenses/Edit.js

```diff
@@ -0,0 +1,22 @@
+import PropTypes from 'prop-types';
```

Comment:
I feel like the only components that should be in `containers` directory are those that are connected via redux. (that's my understanding of the organizational design behind container vs component directories.)

with non-connected components now in `containers` dir, it seems less clear where to put things

Reply:
Ah, I didn't know that's where the naming came from.  I was operating under the assumption that a container was a thing that wasn't generally reused; that is, essentially a standalone thing that only exists in one context.

Reply:
here's the article that influenced my thinking on this - https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0 (and redux example apps - https://github.com/reactjs/redux/tree/master/examples)

Reply:
So a short summary of the distinction between the two might be:

Components are:
- not connected directly to app state
- do not mutate the state
- receive their data and event handlers via props

Containers are:
- basically every other piece of UI

Is that roughly right?  I can be on board with that.

Reply:
yes, i agree!